src/objects.c

```diff
@@ -835,14 +835,14 @@ static Obj FuncMakeImmutable(Obj self, Obj obj)
 // being printed or viewed to trigger the use of ~ when needed.
 static inline UInt IS_ON_PRINT_STACK(const ObjectsModuleState * os, Obj obj)
 {
-  UInt i;
-  if (!(FIRST_RECORD_TNUM <= TNUM_OBJ(obj)
-        && TNUM_OBJ(obj) <= LAST_LIST_TNUM))
+    UInt i;
+    if (!(FIRST_RECORD_TNUM <= TNUM_OBJ(obj) &&
+          TNUM_OBJ(obj) <= LAST_LIST_TNUM))
+        return 0;
+    for (i = 0; i < os->PrintObjDepth; i++)
+        if (os->PrintObjThiss[i] == obj)
```

Comment:
Note that thanks to the refactoring, we can rewrite this loop to be sane again :-). And the "fix" from 712a56f1138ad5a78db97d4f3a12dafe838267d8 becomes unnecessary.